src/lib/markbind/src/parser.js

```diff
@@ -415,27 +246,21 @@ class Parser {
           reject(err);
           return;
         }
-        const parentSitePath = urlUtils.getParentSiteAbsolutePath(file, config.rootPath, config.baseUrlMap);
-        const userDefinedVariables = config.userDefinedVariablesMap[parentSitePath];
-        const pageVariables = Parser.extractPageVariables(file, data, userDefinedVariables, {});
-        let fileContent = njUtil.renderRaw(data, {
-          ...pageVariables,
-          ...userDefinedVariables,
-        }, {
-          path: actualFilePath,
-        });
-        this._extractInnerVariables(fileContent, context, config);
-        const innerVariables = this.getImportedVariableMap(context.cwf);
-        fileContent = njUtil.renderRaw(fileContent, {
-          ...userDefinedVariables, ...innerVariables,
-        });
+        const { variablePreprocessor } = config;
+
+        const outerFileContentRendered = variablePreprocessor.renderOuterVariables(data, file);
```

Comment:
preview of the next part of this PR: reducing boilerplate for rendering site and page variables together.
hopefully this breaks the pr into two more reviewable components

Only site variables are completed for now, but the aim is to get this chunk down even further to 1 - 3 lines of code